src/packagedcode/pypi.py

```diff
@@ -105,6 +105,23 @@ def recognize(cls, location):
     def compute_normalized_license(self):
         return compute_normalized_license(self.declared_license)
 
+    def repository_homepage_url(self, baseurl=default_web_baseurl):
+        if not self.name:
+            return
+        return '{}/project/{}'.format(baseurl, self.name)
+
+    def repository_download_url(self, baseurl=default_download_baseurl):
+        if not self.name or not self.version:
+            return
+        return '{baseurl}/{name[0]}/{name}/{name}-{version}.tar.gz'.format(baseurl=baseurl, name=self.name, version=self.version)
```

Comment:
Don’t assume there will always be a tarball available. It’s best to use the API and get the full download URL with the correct extension.